extensions/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientRecorder.java

```diff
@@ -24,8 +30,12 @@ public void setSslEnabled(boolean sslEnabled) {
         RestClientBuilderImpl.setSslEnabled(sslEnabled);
     }
 
-    public void initializeResteasyProviderFactory(boolean useBuiltIn, Set<String> providersToRegister,
+    public void initializeResteasyProviderFactory(BeanContainer beanContainer,
+            List<Function<Object, Object>> propertyUnwrappers, boolean useBuiltIn,
+            Set<String> providersToRegister,
             Set<String> contributedProviders) {
+        QuarkusInjectorFactory.CONTAINER = beanContainer;
```

Comment:
We should probably move this (and the code from `ResteasyServerCommonRecorder#setupIntegration()`) to the `ResteasyCommonProcessor`. Actually, we should probably move the `ResteasyServerCommonProcessor#setupInjection()` method and this build step should depend on `ResteasyInjectionReadyBuildItem`, or?

Reply:
Interesting idea, I like it. @gsmet do you agree?

Reply:
In fact, `ResteasyInjectionReadyBuildItem` could hold a `RuntimeValue<org.jboss.resteasy.spi.InjectorFactory>` reference and we could keep the `QuarkusInjectorFactory` fields package-private.

Reply:
Should be fixed now